packages/@aws-cdk/aws-pipes-sources-alpha/lib/kinesis.ts

```diff
@@ -0,0 +1,186 @@
+import { IPipe, ISource, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
+import { Duration } from 'aws-cdk-lib';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { IStream } from 'aws-cdk-lib/aws-kinesis';
+import { DeadLetterConfigParameters } from './deadLetterConfig';
+
+/**
+ * Parameters for the Kinesis source.
+ */
+export interface KinesisSourceParameters {
+  /**
+    * The maximum number of records to include in each batch.
+    *
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-batchsize
+    * @default 1
+    */
+  readonly batchSize?: number;
+
+  /**
+    * Define the target queue to send dead-letter queue events to.
+    *
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-deadletterconfig
+    * @default no dead letter queue
+    */
+  readonly deadLetterConfig?: DeadLetterConfigParameters;
+
+  /**
+   * The maximum length of a time to wait for events.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumbatchingwindowinseconds
+   * @default no batching window
+   */
+  readonly maximumBatchingWindow?: Duration;
+
+  /**
+   * (Streams only) Discard records older than the specified age. The default value is -1, which sets the maximum age to infinite. When the value is set to infinite, EventBridge never discards old records.
+   *
+   * Leave undefined to set the maximum record age to infinite.
```

Comment:
The linter wants this a `Duration`, not a `number`, which makes sense. However, `-1` is an allowed value, which doesn't work with `Duration`. I added this note but open to ideas if there is a better way.

Reply:
I can't find a better solution than:

`@default undefined - If <field> is not set, the duration will be infinite`